controllers/nodehealthcheck_controller.go

```diff
@@ -458,15 +483,19 @@ func (r *NodeHealthCheckReconciler) patchStatus(nhc, nhcOrig *remediationv1alpha
 		nhc.Status.Reason = "NHC is enabled, no ongoing remediation"
 	}
 
-	// skip when no changes
-	if reflect.DeepEqual(nhc.Status, nhcOrig.Status) {
+	mergeFrom := client.MergeFrom(nhcOrig)
```

Comment:
IIUC previously we've skipped the merge in case there are no changes , and now we don't  :thinking: 
What is the motivation for this change ?

Reply:
we still skip it, see lines below.
reflect.DeepEqual didn't work well, I saw tons of patches executed, without any visible change. Now I use the patch feature, wich is also used by actual patch command, to see if there are any changes.